tests/fuzz/fuzz_rpc/fuzz_rpc.cpp

```diff
@@ -0,0 +1,153 @@
+#include "initialisation.h"
+#include "rpc_endpoints.h"
+#include <fuzzer/FuzzedDataProvider.h>
+
+#include <csignal>
+#include <csetjmp>
+#include <iostream>
+#include <vector>
+#include <algorithm>
+
+std::vector<size_t> historical_counts;
+
+// Helper definition for timeout recovery
+static sigjmp_buf jump_env;
+void timeout_handler(int sig) {
+  if (tools::performance_timers) {
+    tools::performance_timers->clear();
+    delete tools::performance_timers;
+    tools::performance_timers = nullptr;
+  }
+
+  siglongjmp(jump_env, 1);
+}
+
+// Helper function to generate a biased selector targeting rarely-fuzzed RPC endpoints
+int biased_selector(FuzzedDataProvider& provider) {
```

Comment:
Why randomly select an endpoint? Don't quite understand this, a simple loop through would be better, no?

Reply:
I think I figured it out. Each endpoint could alter state. So you need to randomly select endpoints to get a better fuzz.